consensus/state.go

```diff
@@ -739,12 +767,13 @@ func (cs *ConsensusState) EnterPrevoteWait(height int, round int) {
 // Enter: any +2/3 precommits for next round.
 // Lock & precommit the ProposalBlock if we have enough prevotes for it,
 // else, unlock an existing lock and precommit nil if +2/3 of prevotes were nil,
-// else, precommit locked block or nil otherwise.
+// else, precommit nil otherwise.
+// NOTE: we don't precommit our locked block (unless theres another POL for it) because it complicates unlocking and accountability
```

Comment:
just because I don't know whether it complicates things or just completely breaks things.
otherwise nothing really wrong with it.